src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -161,67 +161,59 @@ def bulk_record(
             }
         }
         """
-        use_case_id = next(iter(strings.keys()))
-        use_case_path_key = self._get_metric_path_key(strings.keys())
         with writes_limiter.check_write_limits(db_write_keys) as writes_limiter_state:
-            # After the DB has successfully committed writes, we exit this
-            # context manager and consume quotas. If the DB crashes we
-            # shouldn't consume quota.
-            use_case_collection = writes_limiter_state.accepted_keys
-            # TODO: later we will use the whole use case collection instead
-            # of pulling out the key collection
-            filtered_db_write_keys = use_case_collection.mapping[use_case_id]
             del db_write_keys
 
             rate_limited_key_results = UseCaseKeyResults()
+            accepted_keys = writes_limiter_state.accepted_keys
+
             for dropped_string in writes_limiter_state.dropped_strings:
-                key_result = dropped_string.key_result
                 rate_limited_key_results.add_use_case_key_result(
-                    UseCaseKeyResult(
-                        use_case_id, key_result.org_id, key_result.string, key_result.id
-                    ),
+                    use_case_key_result=dropped_string.use_case_key_result,
                     fetch_type=dropped_string.fetch_type,
                     fetch_type_ext=dropped_string.fetch_type_ext,
                 )
 
-            if filtered_db_write_keys.size == 0:
+            if accepted_keys.size == 0:
                 return db_read_key_results.merge(rate_limited_key_results)
 
-            if use_case_path_key is UseCaseKey.PERFORMANCE:
+            table = self._get_table_from_metric_path_key(metric_path_key)
+
+            if metric_path_key is UseCaseKey.PERFORMANCE:
```

Comment:
could this be in `MetricsIngestConfiguration` so we're not coding to specific use-cases e.g.

```
if ingest_config.table_supports_multiple_use_cases():
   new_logic();
```

Reply:
We can, do you prefer that change be made in a separate pr or this one?

Reply:
I guess this already was like that so it can be in a follow-up, but let's not introduce new conditions like this (start with a PR that creates the config, then the change that relies on the config)